papermario/src/code_fa4c0_len_3bf0.c

```diff
@@ -2,35 +2,130 @@
 
 INCLUDE_ASM(code_fa4c0_len_3bf0, func_802D5B10);
 
-INCLUDE_ASM(code_fa4c0_len_3bf0, FadeOutMusic);
+s32 FadeOutMusic(script_context* script, s32 initialCall) {
+    s32* ptrReadPos = script->ptrReadPos;
+    s32 itemID = get_variable(script, *ptrReadPos++);
+    s32* ptrNextPos = ptrReadPos++;
 
-INCLUDE_ASM(code_fa4c0_len_3bf0, SetMusicTrack);
+    return (set_music_track(itemID, -1, 0, get_variable(script, *ptrNextPos++), 8) != 0) * 2;
+}
+
+s32 SetMusicTrack(script_context* script, s32 initialCall) {
+    s32* ptrReadPos = script->ptrReadPos;
+    s32 musicPlayer = get_variable(script, *ptrReadPos++);
+    s32 songID = get_variable(script, *ptrReadPos++);
+    s32 variation = get_variable(script, *ptrReadPos++);
+    s32 volume = get_variable(script, *ptrReadPos++);
+
+    return (set_music_track(musicPlayer, songID, variation, 0x1F4, (volume << 0x10) >> 0x10) != 0) * 2;
```

Comment:
I'd try removing the << 0x10 >> 0x10 and seeing what happens. It may be a cast instead or it may not be necessary.